src/components/organisms/settings/user/organizations/table.tsx

```diff
@@ -43,33 +41,35 @@ export const UserOrganizationsTable = () => {
 				}),
 				type: "error",
 			});
-		}
 
+			return;
+		}
 		addToast({
 			message: t("table.messages.organizationDeleted", { name: organization.displayName }),
 			type: "success",
 		});
-
-		if (!deletingCurrentOrganization) return;
-		setTimeout(async () => {
-			if (!user?.defaultOrganizationId) {
-				LoggerService.error(
-					namespaces.ui.organizationTableUserSettings,
-					t("errors.defaultOrganizationIdMissing", { userId: user?.id })
-				);
-				logoutFunction(true);
-				return;
-			}
-			navigate(`/switch-organization/${user.defaultOrganizationId}`);
-		}, 3000);
 	};
 
-	const isNameInputDisabled = (organizationId: string, amIadminCurrentOrganization?: boolean): boolean =>
-		!!(
-			isLoading.updatingOrganization ||
-			user?.defaultOrganizationId === organizationId ||
-			!amIadminCurrentOrganization
-		);
+	const onClickDeleteOrganization = async (organization: EnrichedOrganization) => {
+		const result = await handleDeleteOrganization(organization);
```

Comment:
you need to pass here organizationId only, you don't need other properties of enriched organization.
```suggestion
		const result = await handleDeleteOrganization(organization.id);
```

Reply:
changed